src/Network/Request.php

```diff
@@ -1375,6 +1381,42 @@ public function setInput($input)
     }
 
     /**
+     * Update the request with a new request data element.
+     *
+     * Returns an updated request object. This method returns
+     * a *new* request object and does not mutate the request in-place.
+     *
+     * @param string $name The dot separated path to insert $value at.
+     * @param mixed $value The value to insert into the request data.
+     * @return self
```

Comment:
`$this` is correct. `self` is not a type hint in phpdoc. I prefer `@return Request` but that's just me. I've found PHPStorm has had issues with `$this` on return types, but in the latest builds it has improved support. It gets confused when there are multiple inheritance or overloaded methods.